### PR TITLE
Tab discarding/Tab strip

### PR DESCRIPTION
**Pull request by bridiver**
add tab discard support through chromium tab manager
also adds browser/tabstrip support for better src/chrome code compatibility
dep for https://github.com/brave/browser-laptop/issues/2388

### CODE REVIEW

**Comment by bridiver** [comment]
tab discarding works by creating a new empty webcontents and copying the navigation controller state from the current tab. It then detaches the old webcontents from the webview and attaches the new one. When the tab is activated it triggers a reload

File: atom/browser/common_web_contents_delegate.cc [hunk 1]
```diff
@@ -208,6 +209,8 @@ void CommonWebContentsDelegate::SetOwnerWindow(
   int32_t id =
       api::Window::TrackableObject::GetIDFromWrappedClass(owner_window);
   if (id > 0) {
+    SessionID sessionID;
```
Comment (darkdh) [inline atom/browser/common_web_contents_delegate.cc]: Is `sessionID` not being used? `TabHelper::SetWindowId` will eventually set SessionID and use it so I think we can remove it here.

**Comment by bridiver** [comment]
It is used right below your comment

**Comment by darkdh** [comment]
no I mean it should be used like [SessionTabHelper::FromWebContents(web_contents())->SetWindowID(session);](https://github.com/brave/muon/pull/173/commits/8d45b46a65f22ed930885866164b5a5482d1ff2e#diff-ec092260e71b3de62ef14e0174de4280R76) because `set_id` only store id in its private attribute

**Comment by bridiver** [comment]
If we let it auto set it will have the wrong value

File: chromium_src/BUILD.gn [hunk 2]
```diff
@@ -417,3 +394,135 @@ source_set("browser") {
...
+source_set("tab_manager") {
+  configs += [ ":chromium_src_config" ]
+  include_dirs = [ "." ] # force this to appear before the chromium src dir
+  sources = [
+    "//chrome/browser/engagement/site_engagement_metrics.cc",
+    "//chrome/browser/engagement/site_engagement_metrics.h",
+    "//chrome/browser/engagement/site_engagement_score.cc",
+    "//chrome/browser/engagement/site_engagement_score.h",
+    "chrome/browser/engagement/site_engagement_service.cc",
+    "chrome/browser/lifetime/application_lifetime.cc",
```
Comment (bbondy) [inline chromium_src/BUILD.gn]: I don't see this file

**Comment by bbondy** [comment]
++

**Pull request merged**